src/SqlPersistence/Config/SqlDialect_MsSqlServer.cs

```diff
@@ -34,7 +34,18 @@ internal override void SetParameterValue(DbParameter parameter, object value)
                 if (value is ArraySegment<char> charSegment)
                 {
                     parameter.Value = charSegment.Array;
-                    parameter.Size = charSegment.Count;
+
+                    // Set to 4000 or -1 to improve query execution plan reuse
+                    // Must be when exceeding 4000 characters for nvarchar(max)  https://stackoverflow.com/a/973269/199551
```

Comment:
Some text is missing in the comment. Is this what you meant?

```suggestion
                    // Must be set when exceeding 4000 characters for nvarchar(max)  https://stackoverflow.com/a/973269/199551
```